clients/web/src/app.js

```diff
@@ -1,43 +1,82 @@
 girder.App = girder.View.extend({
-    initialize: function () {
-        girder.fetchCurrentUser()
-            .done(_.bind(function (user) {
-                girder.eventStream = new girder.EventStream({
-                    timeout: girder.sseTimeout || null
-                });
+    /**
+     * @param {object} [settings]
+     * @param {bool} [settings.start=true] Run start after initialization
+     */
+    initialize: function (settings) {
+        this._started = false;
+        settings = settings || {};
+        if (settings.start === undefined || settings.start) {
+            this.start();
+        }
+    },
 
-                this.headerView = new girder.views.LayoutHeaderView({
-                    parentView: this
-                });
+    /**
+     * Start the application with optional components.
+     * @param {object} [settings]
+     * @param {bool} [settings.fetch=true] Fetch the current user modal
+     * @param {bool} [settings.render=true] Render the layout after starting
+     * @param {bool} [settings.history=true] Start backbone's history api
+     * @returns {$.Deferred} A promise-like object that resolves when the app is ready
+     */
+    start: function (settings) {
+        // start is a noop if the app is already running
+        var promise = new $.Deferred().resolve(null).promise();
+        if (this._started) {
+            return promise;
+        }
+        this._started = true;
 
-                this.globalNavView = new girder.views.LayoutGlobalNavView({
-                    parentView: this
-                });
+        // set defaults
+        settings = _.defaults(settings || {}, {
+            fetch: true,
+            render: true,
+            history: true
+        });
 
-                this.footerView = new girder.views.LayoutFooterView({
-                    parentView: this
-                });
+        // define a function to be run after fetching the user model
+        var afterFetch = _.bind(function (user) {
+            girder.eventStream = new girder.EventStream({
+                timeout: girder.sseTimeout || null
+            });
 
-                this.progressListView = new girder.views.ProgressListView({
-                    eventStream: girder.eventStream,
-                    parentView: this
-                });
+            this._createLayout();
 
-                if (user) {
-                    girder.currentUser = new girder.models.UserModel(user);
-                    girder.eventStream.open();
-                }
+            if (user) {
+                girder.currentUser = new girder.models.UserModel(user);
+                girder.eventStream.open();
```

Comment:
I thought about adding more options, but it gets dicey.  For example, if the event stream object is assumed to exist all over the place, including in the `_createLayout` function.  I think if you really want to do that, you will have to override the `start` method anyway.